### ISSUE TITLE
Data Colors reset on each filter (when using Legend)

### ISSUE THREAD

**Issue by Mike-Honey** [Microsoft/powerbi-visuals-drilldown-choropleth#6]
When using the Drill Down Choropleth visual with a Legend, you can set Data Colors in the normal manner.  However (unlike most visuals) these dont "stick" - every time the visual is filtered it re-assigns colors from the default palette.

You can repro this behaviour using the sample PBIX file provided with the visual.  From the page: Choropleth with Legend, go to the Visual Level Filters and filter the Party field, e.g. select B and D.  You can also try using the Format / Data Colors property to set a specific color, then change the filter and watch it reset to the default palette.

For my scenario, each value in the Legend field should show in a specific colour, regardless of filtering.  This is achievable with other visuals e.g. TreeMap.  Please fix this aspect of an otherwise outstanding Custom Visual.

**Comment by Mike-Honey** [comment]
FWIW, any help with this issue will contribute in a small part to literally saving the planet – the visual is to help the people running this project:
https://www.moore.org/initiative-strategy-detail?initiativeId=andes-amazon-initiative

**Comment by Mike-Honey** [comment]
Great - thanks!

**Comment by nuwanu** [comment]
I have fixed the issue using code and now color is not duplicate in Legend if anyone need help let me know

**Comment by nikhilkabbin** [comment]
@nuwanu eager to see the fix. Can you point me to the it please?

**Comment by MutantApe** [comment]
> I have fixed the issue using code and generate new plugin and upload to power bi as Organization plugin now color is not duplicate in Legends. if anyone need help let me know

Need help in stacked bar chart with codes for the same issue.

**Comment by lerayume** [comment]
Yeah, I feel like it will actually never be fixed by the creators as we are in 2021 and still the problem is there. I am not that good at coding as @nuwanu, so I can't fix the issue myself and I can't either filter for all the thousands of filtering possibilities to fix the colors every time. Crazy!

**Comment by kadirkkkk** [comment]
> 
> 
> I have fixed the issue using code and generate new plugin and upload to power bi as Organization plugin now color is not duplicate in Legends. if anyone need help let me know

Can you help me please
